### PR TITLE
Ivy publication resolved versions

### PR DESCRIPTION
**Pull request by rpalcolea**
This uses https://docs.gradle.org/current/userguide/publishing_ivy.html#publishing_ivy:resolved_dependencies introduced in Gradle 5.4

Few things to mention:

* `revConstraint` is only informative and Gradle doesn't have support for this in ivy descriptor with dynamic dependencies. There's PR and discussion in https://github.com/gradle/gradle/pull/9121/files
* Our plugin used to reflect replaced modules in ivy descriptor. Gradle doesn't support this directly but it isn't a show stopper for us. I'll open an issue soon with more details 

Sample resolved dependencies:

```
    <dependency org="test" name="foo" rev="1.0.0" conf="compile-&gt;default"/>
    <dependency org="test" name="foo" rev="1.0.0" conf="runtime-&gt;default"/>
    <dependency org="test" name="foo" rev="1.0.0" conf="compile-&gt;default"/>

```

### CODE REVIEW

File: src/main/groovy/nebula/plugin/publishing/ivy/IvyResolvedDependenciesPlugin.groovy [hunk 1]
```diff
@@ -84,25 +45,21 @@ class IvyResolvedDependenciesPlugin extends AbstractResolvedDependenciesPlugin {
...
-            dep.@rev = selected.version
-        }
-    }
-
-    private void updateReplacedModules(ModuleVersionIdentifier mvid, String group, String name, Node dep) {
-        if (mvid.group != group || mvid.name != name) {
-            dep.@org = mvid.group
-            dep.@name = mvid.name
-            dep.@rev = mvid.version
+    private void verifyFirstDependencyExclusions(Project project) {
```
Comment (chali) [inline src/main/groovy/nebula/plugin/publishing/ivy/IvyResolvedDependenciesPlugin.groovy]: What case is this verification catching? Do we have a test?

File: src/test/groovy/nebula/plugin/publishing/ivy/interaction/IvyPublishRecommenderInteractionSpec.groovy [hunk 2]
```diff
@@ -18,7 +18,7 @@ class IvyPublishRecommenderInteractionSpec extends IntegrationTestKitSpec {
 
         buildFile << """\
             plugins {
-                id 'java'
+                id 'java-library'
```
Comment (chali) [inline src/test/groovy/nebula/plugin/publishing/ivy/interaction/IvyPublishRecommenderInteractionSpec.groovy]: Should we keep both flavors? `java` and `java-library`?

File: src/main/groovy/nebula/plugin/publishing/ivy/IvyResolvedDependenciesPlugin.groovy (hunk 1 above)
Comment (rpalcolea) [reply src/main/groovy/nebula/plugin/publishing/ivy/IvyResolvedDependenciesPlugin.groovy]: we do, here is the test: https://github.com/nebula-plugins/nebula-publishing-plugin/blob/95713c49a04ebbad4ff78571687e1e6369614997/src/test/groovy/nebula/plugin/publishing/ivy/IvyResolvedDependenciesPluginIntegrationSpec.groovy#L177

Comment (chali) [reply src/main/groovy/nebula/plugin/publishing/ivy/IvyResolvedDependenciesPlugin.groovy]: Oh I see it was originally in the parent class.

File: src/test/groovy/nebula/plugin/publishing/ivy/interaction/IvyPublishRecommenderInteractionSpec.groovy (hunk 2 above)
Comment (rpalcolea) [reply src/test/groovy/nebula/plugin/publishing/ivy/interaction/IvyPublishRecommenderInteractionSpec.groovy]: sure it doesn't hurt, added it in https://github.com/nebula-plugins/nebula-publishing-plugin/pull/129/commits/e18c9527cef9bae62510efde2ad47e158e3cc3d4

File: src/main/groovy/nebula/plugin/publishing/ivy/IvyResolvedDependenciesPlugin.groovy (hunk 1 above)
Comment (rpalcolea) [reply src/main/groovy/nebula/plugin/publishing/ivy/IvyResolvedDependenciesPlugin.groovy]: yeah! sorry, forgot to mention that!

**Pull request merged**